File: house-robber/house-robber.java
class Solution {
    public int rob(int[] nums) {
        
        int  firstRob = 0;
        int secondRob = 0;
        int max = 0;
        for(int pointer: nums){
            max = Math.max(firstRob + pointer, secondRob);
            firstRob = secondRob;
            secondRob = max;
        }
        return secondRob;
        
    }
}

File: house-robber/house-robber.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        
        first = 0
        second = 0
        maxRob = 0
        
        for i in nums:
            maxRob = max(first + i, second)
            first = second
            second = maxRob
            
        return maxRob
        